### PR TITLE
✨ Add log relaying to requester

### LINKED ISSUE #159: Extend requester to accept relay of vLLM instance log
This is part of #157

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
This PR extends the requester container to accept log chunks.

Resolves #159

### CODE REVIEW

File: pkg/server/requester/coordination/server.go [hunk 1]
```diff
@@ -143,13 +144,66 @@ func newSetReadyHandler(logger klog.Logger, ready *atomic.Bool, newReady bool) f
...
+		}
+		news, err := io.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			_, _ = fmt.Fprintf(w, "Failed to read log chunk from request body: %s", err.Error())
+			return
+		}
+		if startPos+int64(len(news)) <= nextLogPos {
+			w.WriteHeader(http.StatusOK)
+			_, _ = fmt.Fprintf(w, "Accepted startPos=%d, chunkLen=%d, but that has nothing new; next pos=%d\n", startPos, len(news), nextLogPos)
```
Comment (diegocastanibm) [inline pkg/server/requester/coordination/server.go]: I prefer to use `contentLen` as you use below. Also, it would be good to be consistent with the terminology in the response messages

File: pkg/server/requester/coordination/server_test.go [hunk 2]
```diff
@@ -74,3 +77,60 @@ func FuzzServer(f *testing.F) {
 	})
 	cancel()
 }
+
+func TestLogChunking(t *testing.T) {
+	var rightLog [600]byte
```
Comment (diegocastanibm) [inline pkg/server/requester/coordination/server_test.go]: can you name `600` as a constant with some explanation?

File: pkg/server/requester/coordination/server_test.go [hunk 3]
```diff
@@ -74,3 +77,60 @@ func FuzzServer(f *testing.F) {
...
+	for idx := range 600 {
+		rightLog[idx] = '0' + byte(rand.IntN(10))
+	}
+	var logBuilder strings.Builder
+	ctx, cancel := context.WithCancel(t.Context())
+	var ready atomic.Bool
+	port := "28084"
```
[the remark this answers is not in the document]
Comment (diegocastanibm) [reply pkg/server/requester/coordination/server_test.go]: OK, but please, could you create an issue and link this comments so we do not forget about it?

File: pkg/server/requester/coordination/server.go (hunk 1 above)
Comment (MikeSpreitzer) [reply pkg/server/requester/coordination/server.go]: I see only "chunkLength" in https://github.com/MikeSpreitzer/llm-d-fast-model-actuation/blob/8e9dea2680d4ee1c047922fbf7b4a7b0af3f4e98/pkg/server/requester/coordination/server.go

**Comment by MikeSpreitzer** [comment]
The latest commits add testing of empty chunks and negative starting positions. I am not clear on what else is needed.

**Comment by diegocastanibm** [comment]
> The latest commits add testing of empty chunks and negative starting positions. I am not clear on what else is needed.

These could be some of the boundary cases to test:
startPos = 0, chunkLen = 0 (empty write at beginning)
startPos = curLen, chunkLen = 1 (exactly one byte forward)
startPos = curLen, chunkLen = 0 (empty write at exact current position)
startPos = curLen - 1, chunkLen = 1 (exactly overwrites last byte)
startPos = curLen - 1, chunkLen = 2 (overwrites last byte + adds one new)
Chunk that exactly fills to fullLogSize (600 bytes)
What do you think?
